docs/adr/0003-tarball-contents.md

```diff
@@ -0,0 +1,70 @@
+
+Date: 2020-11-23
+
+## Status
+
+proposed
+
+## Context
+
+The release pipeline of Wikibase will produce a tarball release package containing a tested version of the Wikibase extension.
+The package will normally include the necessary files to run the software and sometimes documentation of changes to the software and how to use them.
+
+In order to get a overview on the practices by the open-source php community a brief survey was done on the releases of some popular projects.
+
+
+| Name      | Release notes  |   Historic release notes    | Installation instructions  | Upgrade instructions | Hidden files  | Vendor folder |
+| --------- | ---------------| --------------------------- | -------------------------- | ---------------------| --------------|---------------|
+| mediawiki |      yes       |            yes              |            yes             |          yes         |      no       |       no
+| laravel   |      no (1)    |            no               |            no              |          no          |      some (6) |       no
+| nextcloud |      no (2)    |            no               |            no              |          no          |      some (7) |       yes
+| joomla    |      no (3)    |            no (3)           |            yes (3)         |          no (4)      |      no       |       yes
+| wordpress |      no (5)    |            no               |            yes (8)         |          no          |      no       |       -
+
+---
+
+1. Online https://laravel.com/docs/5.8/releases
+2. Online https://nextcloud.com/changelog/
+3. Installation/INSTALL & simple readme with links to https://docs.joomla.org/Special:MyLanguage/Joomla_3.9_version_history, https://downloads.joomla.org/technical-requirements
+4. Specific upgrade release packages
+5. Online https://wordpress.org/news/2020/11/wordpress-5-6-beta-4/
+6. gitignore, editorconfig etc.
+7. Necessary htaccess files
+8. Readme.html with installation & upgrade instructions and links to documentation for further reading.
+
+---
+
+MediaWiki sticks out from the crowd by including most of their release specific documentation in each of the tarball releases while all other projects in the survey do not. At first glance this feels like a undesirable pattern to adopt as it's uncommon among other projects. In reality this points out a tested and proven methodology for documenting changes and shipping release notes by the mediawiki community. Release notes are added to the release branch and refined throughout the release process and then used as a template for the documentation on mediawiki.org.     
+## Decision
+
+Seeing that there is an already defined process for release notes by the mediawiki community it feels natural to use existing tools and adopting this process.
+
+The `vendor` folder will not be included in the tarball as composer dependencies should be merged by the [composer merge plugin](https://github.com/wikimedia/composer-merge-plugin).  
```

Comment:
I'm still really unsure about this:
For example [the mediawiki installation requirements](https://www.mediawiki.org/wiki/Manual:Installation_requirements) don't include composer and [Installing mediawiki for developers](https://www.mediawiki.org/wiki/Manual:Installing_MediaWiki#Git_(for_developers)) seems to imply composer shouldn't need to be run when "just normally" installing

Reply:
[Wikibase install instructions](https://www.mediawiki.org/wiki/Wikibase/Installation) however heavily reference composer

Reply:
From https://www.mediawiki.org/wiki/Manual:External_libraries:
> The extension distributor automatically packages composer dependencies, so tarball users won't have to do it manually.

`ExtensionProcessor::getExtraAutoloaderPaths` also seems to pick up these vendor folders… so I guess we might want to revise this? But I'm not sure how well this copes with conflicting dependency versions (I guess not well/ at all).

Reply:
From https://www.mediawiki.org/wiki/Manual:External_libraries:
> The extension distributor automatically packages composer dependencies, so tarball users won't have to do it manually.

`ExtensionProcessor::getExtraAutoloaderPaths` also seems to pick up these vendor folders… so I guess we might want to revise this? But I'm not sure how well this copes with conflicting dependency versions (I guess not well/ at all).

Reply:
I poked the release team about their reasoning for including the `vendor` folder in and the arguments were the following

- Not everyone can run composer (security, shared hosting, shell access etc)
- Including it we can support both people that has access to composer and people that don't (It's easier to rm -rf vendor and the composer.lock (if you don't need them) than it is to add them later)

original task: https://phabricator.wikimedia.org/T70940

I can agree this makes sense to me. But we would probably need to update the documentation as you say @tarrow.

Reply:
I did a poor job at investigating the mediawiki tarballs - they indeed include the vendor.